src/helpers/graph.js

```diff
@@ -0,0 +1,22 @@
+export function addGroupTagToNotes(notes, edges) {
+    if (notes && edges) {
+        notes = notes.map(note => { // add group tag to all the egdes
+            note.group = edges.filter(edge => edge.from === note.id).length > 0 ? 'parent' : 'children' 
+            return note
+        })
+    }
+    return notes
+}
+
+export function removeNoteFromGraph(notes, edges, noteToDelete) {
+    notes = notes.filter(note => note.id !== noteToDelete.id) // remove the note from the array
+    edges = edges.filter(edge => {  // remove all the children of the deleted parent
+      if (edge.from === noteToDelete.id) {
```

Comment:
I think you're deleting only the immediate children of a node (first level), omitting the nested children.

I.e. I don't see this function deleting children of children of a node.

It is not deleting the whole subtree, it is deleting only the first level of the subtree. If I understood the code correctly.